File: backend/app/schemas/delegacao.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, EmailStr, Field


class DelegacaoInviteRequest(BaseModel):
    email: EmailStr
    can_write: bool = True


class DelegacaoConfirmRequest(BaseModel):
    nome: Optional[str] = None
    password: Optional[str] = Field(default=None, min_length=6)


class UserResumo(BaseModel):
    id: int
    nome: str
    email: EmailStr

    class Config:
        from_attributes = True


class DelegacaoResponse(BaseModel):
    id: int
    owner_user_id: int
    delegate_user_id: Optional[int]
    invited_email: EmailStr
    status: str
    can_write: bool
    invite_expires_at: Optional[datetime] = None
    created_at: Optional[datetime] = None
    accepted_at: Optional[datetime] = None
    revoked_at: Optional[datetime] = None
    owner: UserResumo
    delegate: Optional[UserResumo] = None

    class Config:
        from_attributes = True


class DelegacaoInviteResponse(BaseModel):
    delegacao: DelegacaoResponse
    has_account: bool
    email_sent: bool


class DelegacaoContextOption(BaseModel):
    user_id: int
    nome: str
    email: EmailStr
    can_write: bool
    is_owner: bool


class DelegacaoInviteTokenInfo(BaseModel):
    invited_email: EmailStr
    owner_nome: str
    owner_email: EmailStr
    has_account: bool
    expired: bool

File: frontend/src/types/delegacao.ts
import type { User } from './auth'

export interface DelegacaoContextOption {
  user_id: number
  nome: string
  email: string
  can_write: boolean
  is_owner: boolean
}

export interface Delegacao {
  id: number
  owner_user_id: number
  delegate_user_id?: number | null
  invited_email: string
  status: 'pending' | 'active' | 'revoked'
  can_write: boolean
  invite_expires_at?: string | null
  created_at?: string
  accepted_at?: string
  revoked_at?: string
  owner?: User
  delegate?: User | null
}

export interface DelegacaoInviteResponse {
  delegacao: Delegacao
  has_account: boolean
  email_sent: boolean
}

export interface DelegacaoInviteTokenInfo {
  invited_email: string
  owner_nome: string
  owner_email: string
  has_account: boolean
  expired: boolean
}
